kms/keysource.go

```diff
@@ -77,13 +77,14 @@ type MasterKey struct {
 	epResolver aws.EndpointResolverWithOptions
 }
 
-// NewMasterKey creates a new MasterKey from an ARN, role and context, setting
+// NewMasterKey creates a new MasterKey from an ARN, role, context and awsProfile, setting
 // the creation date to the current date.
-func NewMasterKey(arn string, role string, context map[string]*string) *MasterKey {
+func NewMasterKey(arn string, role string, context map[string]*string, awsProfile string) *MasterKey {
```

Comment:
Think it would be better to introduce a new `NewMasterWithProfile` to not break the signature of this function.